src/dune_rules/expander.ml

```diff
@@ -663,6 +663,13 @@ let expand_pform_macro
            let* artifacts_host = t.artifacts_host in
            let+ b = Artifacts.binary_available artifacts_host s in
            b |> string_of_bool |> string))
+  | File_available ->
+    Direct
+      (Without
+        (let open Memo.O in
+         let path = relative ~source dir s in
+         let+ available = Fs.file_exists path in
```

Comment:
I think you just want `Build_system.file_exists` here. You should never have a source path here. Source paths are transparently copied to _build/default for the rules to consume. If you can observe a source path here, then we have a problem.